bson/src/main/org/bson/codecs/pojo/EnumCodec.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs.pojo;
+
+import org.bson.BsonReader;
+import org.bson.BsonWriter;
+import org.bson.codecs.Codec;
+import org.bson.codecs.DecoderContext;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.StringCodec;
+
+final class EnumCodec<T extends Enum<T>> implements Codec<T> {
+    private final Codec<String> codec = new StringCodec();
+    private final Class<T> clazz;
+
+    EnumCodec(final Class<T> clazz) {
+        this.clazz = clazz;
+    }
+
+    @Override
+    public void encode(final BsonWriter writer, final T value, final EncoderContext encoderContext) {
+        codec.encode(writer, value.toString(), encoderContext);
```

Comment:
I think this should be value.name() instead of value.toString().  If someone inadvertently overrides toString (say to print out the properties of a value), then this won't roundtrip anymore.